lib/pavilion/schedulers/plugins/flux.py

```diff
@@ -0,0 +1,274 @@
+        extra = {"nodes_listing": listing}
+        return nodelist, extra
+
+    def _transform_raw_node_data(self, sched_config, node_data, extra) -> NodeInfo:
+        """
+        Translate the gathered data into a NodeInfo dict.
+        """
+        listing = extra["nodes_listing"]
+        del extra["nodes_listing"]
+        node_info = NodeInfo({})
+        node_info["name"] = node_data
+        node_info["cpus"] = listing.free.ncores
+        node_info["up"] = node_data in list(listing.up.nodelist)
+        node_info["available"] = node_data in list(listing.free.nodelist)
+        return node_info
+
+    def _available(self) -> bool:
+        """
+        Ensure we can import and talk to flux.
+        """
+        return flux is not None
+
+    def _kickoff(self, pav_cfg, job: Job, sched_config: dict) -> JobInfo:
+        """
+        Submit the kick off script using sbatch.
+        """
+
+        output = job.sched_log.as_posix()
+        error = output.replace(".log", ".err")
+
+        # Generate the flux job
+        # Assume the filename includes a hashbang
+        # flux does not support mem_mb, disk_mb
+        fluxjob = JobspecV1.from_command(
+            command=["/bin/bash", job.kickoff_path.as_posix()],
+            num_tasks=sched_config["tasks_per_node"],
+        )
+
+        # Min time is one minute
+        limit = sched_config["time_limit"] or 0
+        limit = limit * sched_config["time_limit"] * 60
+
+        # A duration of zero (the default) means unlimited
+        fluxjob.duration = limit
+        fluxjob.stdout = output
+        fluxjob.stderr = error
+
+        # This doesn't seem to be used?
+        fluxjob.cwd = str(job.tests_path)
+        fluxjob.environment = dict(os.environ)
+
+        # This submits without waiting
+        flux_future = flux.job.submit_async(flux.Flux(), fluxjob)
+
+        jobid = flux_future.get_id()
+        while not jobid:
+            jobid = flux_future.get_id()
+            time.sleep(5)
+
+        job._jobid = str(jobid)
+        job._submit_time = time.time()
+        job._flux_future = flux_future
+        sys_name = sys_vars.get_vars(True)["sys_name"]
+
+        return JobInfo(
+            {
+                "id": str(jobid),
+                "jobid": jobid.orig_str,
+                "sys_name": sys_name,
+                "name": job.name,
+            }
+        )
+
+    def _job_status(self, pav_cfg, job_info: JobInfo) -> TestStatusInfo:
+        """
+        Get the current status of the flux job for the given test.
+        """
+        sys_name = sys_vars.get_vars(True)["sys_name"]
+        if job_info["sys_name"] != sys_name:
+            return TestStatusInfo(
+                STATES.SCHEDULED,
+                "Job started on a different cluster ({}).".format(sys_name),
+            )
+
+        listing = flux.job.list.JobList(flux.Flux(), ids=[job_info["jobid"]])
+        jobs = listing.jobs()
+
+        if not listing:
```

Comment:
Should this be `if not jobs`?

Reply:
Style note - Pavilion has a separate job object, so we should probably be disambiguating them.  Maybe call them `flux_jobs`?

Reply:
yes it should! I renamed / debugged this a bunch of times and forgot to update the name here, should be fixed now.